### DISCUSSION TITLE
Is there a headless mode for Linux machines?

### DISCUSSION THREAD

**Discussion by LeftyBC** [timmo001/system-bridge#360] [Help] [answered]
Just discovered system-bridge, and I'd like to deploy it to my home machines so they can participate in my home automation.  Some of those machines are headless servers with no graphical capabilities (running Fedora server).  Does system-bridge support running without a desktop environment or X11 server running?

**Comment by timmo001** [comment]
There isn't unfortunately. The app relies on electron which requires a desktop environment.

I do think that this could potentially run headlessly though, even if the app is at a more limited functionality (no media player functionality etc.)

**Reply by timmo001** [reply]
Doing this would also help with #393

**Comment by timmo001** [comment, marked as answer]
Recently rewritten the application to not use electron. TBR soon™

**Reply by IamTheCarl** [reply]
Was this the 2.0.0 release?
I tried to get headless mode to launch on a headless system of mine and don't want to open a bug report if this feature is not actually implemented yet.

**Reply by timmo001** [reply]
Not yet. I am currently making improvements to allow for a 'cli' mode in a docker container. Doing so means a few features have to be disabled, such as keyboard, open and other desktop only features.

The UI is also usable, but could also do with a revamp so a link from the tray isn't the only option

**Reply by tscibilia** [reply]
Great work on this project, I recently found it via HomeAssistant. I was wondering if there were any updates on running it headless (I have a proxmox instance I'd like to track). If so, do you have any documentation on setting it up? I wasn't sure if I could just install the .deb package from v.2.5
